publisher-sdk/src/main/java/com/criteo/publisher/util/jsonadapter/BooleanJsonAdapter.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.util.jsonadapter
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonDataException
+import com.squareup.moshi.JsonReader
+import com.squareup.moshi.JsonWriter
+import java.lang.Boolean.parseBoolean
+
+internal class BooleanJsonAdapter : JsonAdapter<Boolean>() {
```

Comment:
Could you add a comment explaining what this adapter is doing and why

Reply:
As far i see from tests, sometimes boolean value comes as string. For example [here](https://github.com/criteo/android-publisher-sdk/blob/3eed94efce7629818211f8a890c8d5d7bef64965/publisher-sdk/src/test/java/com/criteo/publisher/model/CdbResponseSlotTest.java#L579). To not break this logic i added this adapter.